tests/components/nest/test_config_flow_sdm.py

```diff
@@ -34,13 +35,24 @@
 APP_REDIRECT_URL = "urn:ietf:wg:oauth:2.0:oob"
 
 
-def get_config_entry(hass):
+def get_config_entry(hass: HomeAssistant) -> ConfigEntry:
     """Return a single config entry."""
     entries = hass.config_entries.async_entries(DOMAIN)
     assert len(entries) == 1
     return entries[0]
 
 
+def create_config_entry(hass: HomeAssistant, data: dict) -> ConfigEntry:
+    """Create the ConfigEntry."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=data,
+        unique_id=DOMAIN,
```

Comment:
This recreates the existing behaviour, and probably it is low risk during tests, but is `unique_id=DOMAIN` really what you want?  That won't be unique if you call it twice.

Reply:
As in calling the config flow twice?

If I understand you right: Yes, nest currently only supports one ConfigEntry at a time, given the high complexity of everything else on the integration. (It supports two APIs, and 3 authentication methods for example).

Reply:
There is a test below that exercises multiple config entries and I think it does not use this function.

Reply:
Ok, if only one config entry allowed then this probably doesn't matter.  I looked in the tests source for other integrations and found that most cases the unique_id was set to a string representing a MAC address or similar.  I did not see anyone else using DOMAIN, but I did see lots of arbitrary fixed strings (which have the same limitations as a string like DOMAIN).

So probably not worth worrying about.